test/types/virtuals.test.ts

```diff
@@ -95,14 +95,15 @@ async function autoTypedVirtuals() {
   const testSchema = new Schema({
     email: {
       type: String,
-      required: [true, 'email is required']
+      required: [true, 'email is required'] as [true, string]
```

Comment:
are those array castings really required instead of `as const`?

Reply:
Unfortunately yes. I think that's more of a TypeScript 5 issue though, similar to #13514.